### PR TITLE
feat(front): u#2032 t#3724 create status

### PR DESCRIPTION
**Pull request by cocotime**
It has to be tested running `back` on the following branch: us/2032/task/3721/create-status-endpoint

### CODE REVIEW

File: javascript/apps/taiga/src/app/modules/project/feature-kanban/components/create-status/kanban-create-status.component.html [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+            <ng-container inputError>
+              <tg-ui-error error="required">
+                {{ t('kanban.status_empty') }}</tg-ui-error
+              >
+            </ng-container>
+          </tg-ui-input>
+        </div>
+        <div
+          *ngIf="
+            form.get('status')!.value?.length &&
```
Comment (juanfran) [inline javascript/apps/taiga/src/app/modules/project/feature-kanban/components/create-status/kanban-create-status.component.html:42]: I think the first part if not necessary

File: javascript/apps/taiga/src/app/modules/project/feature-kanban/components/create-status/kanban-create-status.component.html [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        <div class="actions">
+          <button
+            tuiButton
+            data-test="status-create"
+            appearance="primary"
+            type="submit">
+            {{ t('kanban.create_status') }}
+          </button>
+
+          <button
```
Comment (juanfran) [inline javascript/apps/taiga/src/app/modules/project/feature-kanban/components/create-status/kanban-create-status.component.html:57]: type button missing

File: javascript/apps/taiga/src/app/modules/project/feature-kanban/components/create-status/kanban-create-status.component.ts [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+  public columnSize = 292;
+  public statusMaxLength = 30;
+
+  public form = this.fb.group({
+    status: [
+      '',
+      [
+        Validators.required,
+        Validators.maxLength(this.statusMaxLength),
+        Validators.pattern(/^(\s+\S+\s*)*(?!\s).*$/),
```
Comment (juanfran) [inline javascript/apps/taiga/src/app/modules/project/feature-kanban/components/create-status/kanban-create-status.component.ts:63]: What this means would be useful

File: javascript/apps/taiga/src/app/modules/project/feature-kanban/components/create-status/kanban-create-status.component.html [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+<ng-container *transloco="let t">
+  <div
+    class="create-status"
+    #createStatusWrapper
+    *hasPermission="['create']; entity: 'story'">
+    <ng-container *ngIf="showAddForm; else addStatusTpl">
+      <form
+        class="form-card"
+        (tuiActiveZoneChange)="leaveForm($event)"
+        [showFormErrors]="submitted"
```
Comment (juanfran) [inline javascript/apps/taiga/src/app/modules/project/feature-kanban/components/create-status/kanban-create-status.component.html:18]: Maybe you can do this? (I'm not 100% sure if it's works in your use case)

```html
      <form
       #form="ngForm"
        class="form-card"
        (tuiActiveZoneChange)="leaveForm($event)"
        [showFormErrors]="form.submitted"
```

**Comment by juanfran** [comment]
When you close the form without saving, the focus should go back to "add status"

**Comment by juanfran** [comment]
When the user leaves the form, the state is created, but the form is still open, is this correct?

**Comment by juanfran** [comment]
The margin right between status & form is more than 8px

File: javascript/apps/taiga/src/app/modules/project/feature-kanban/components/create-status/kanban-create-status.component.html (hunk 1 above)
Comment (cocotime) [reply javascript/apps/taiga/src/app/modules/project/feature-kanban/components/create-status/kanban-create-status.component.html]: deleted

File: javascript/apps/taiga/src/app/modules/project/feature-kanban/components/create-status/kanban-create-status.component.html (hunk 2 above)
Comment (cocotime) [reply javascript/apps/taiga/src/app/modules/project/feature-kanban/components/create-status/kanban-create-status.component.html:52]: added

File: javascript/apps/taiga/src/app/modules/project/feature-kanban/components/create-status/kanban-create-status.component.ts (hunk 3 above)
Comment (cocotime) [reply javascript/apps/taiga/src/app/modules/project/feature-kanban/components/create-status/kanban-create-status.component.ts:70]: comment added

File: javascript/apps/taiga/src/app/modules/project/feature-kanban/components/create-status/kanban-create-status.component.html (hunk 4 above)
Comment (cocotime) [reply javascript/apps/taiga/src/app/modules/project/feature-kanban/components/create-status/kanban-create-status.component.html:15]: It wasn't working

**Review by juanfran — APPROVED** [review]

File: javascript/apps/taiga/src/app/modules/project/feature-kanban/project-feature-kanban.component.ts [hunk 5]
```diff
@@ -261,6 +262,29 @@ export class ProjectFeatureKanbanComponent {
         );
       });
 
+    this.wsService
+      .projectEvents<{ workflowStatus: WorkflowStatus }>(
+        'workflowstatuses.create'
+      )
+      .pipe(untilDestroyed(this))
```
Comment (juanfran) [inline javascript/apps/taiga/src/app/modules/project/feature-kanban/project-feature-kanban.component.ts:269]: Use the new `takeUntilDestroyed` or rx-angular `hold`

Comment (cocotime) [reply javascript/apps/taiga/src/app/modules/project/feature-kanban/project-feature-kanban.component.ts:270]: changed

**Pull request merged**